plugins/coatils.py

```diff
@@ -124,3 +127,116 @@ def ls(self, msg, match):
                 yield ''.join(selected_bears) + ' |'
             else:
                 yield 'No bears found for {}'.format(lang)
+
+    @staticmethod
+    def construct_settings(settings):
+        settings = settings.strip().split()
+
+        def is_setting(x): return '=' in x  # Ignore PycodestyleBear (E731)
+        setting_dict = OrderedDict()
+        for candidate in settings:
+            if not is_setting(candidate):  # bear
+                setting_dict[candidate.strip()] = dict()
+            else:
+                key, value = candidate.strip().split('=')
+                setting_dict[list(setting_dict.keys())[-1]][key] = value
+        return setting_dict
+
+    @staticmethod
+    def position(stl, stc, enl, enc):
+        if stc is None and enc is None:
+            if stl != enl:
+                return 'Between lines {} and {}'.format(stl, enl)
+            else:
+                return 'At line {}'.format(stl)
+        if stc is None and enc is not None:
+            return 'Between line {} and position {}:{}'.format(stl, enl, enc)
+        if stc is not None and enc is None:
+            return 'Between position {}:{} and line {}'.format(stl, stc, enl)
+        if stl == enl:
+            if stc == enc:
+                return "At {}:{}".format(stl, stc)
+            else:
+                return "At line {}, between col {} and {}".format(stl, stc, enc)
+        else:
+            return "Between positions {}:{} and {}:{}".format(stl, stc,
+                                                              enl, enc)
+
+    # Ignore PycodestyleBear, LineLengthBear
+    @re_botcmd(pattern=r'run\s+(\w+)((?:\s+\w+(?:\s+\w+=\w+)*)+)\n+```\n([\s\S]+)\n```')
+    def run(self, msg, match):
+        """
+        Run coala over the given code.
+
+        Example: `run Bear1 setting1=something setting2=something Bear2\ncode`
```

Comment:
Why dont you use the settings parser used for the command line settings argument?